src/main/java/no/training/project/resources/datecalculation/DateConversion.java

```diff
@@ -0,0 +1,78 @@
+package no.training.project.resources.datecalculation;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.QueryParam;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.text.NumberFormat;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+@Path("/date-conversion")
+public class DateConversion {
+    private static final Logger log = LoggerFactory.getLogger(DateConversion.class);
+
+    @GET
+    @Path("/")
+    public long calculateSol(@QueryParam("date") String date) {
+        try {
+            log.info("Given input is {}", date);
+            LocalDate inputDate = null;
+
+            if (date == null) {
+                log.debug("Given input date is null");
+                inputDate = LocalDate.now();
+            } else {
+                log.debug("Given input date is not null");
+                inputDate = LocalDate.parse(date, DateTimeFormatter.ISO_DATE);
+                log.debug("Given input date is parsed to LocalDate");
+            }
+            LocalDate referenceDate = LocalDate.parse("2012-08-06", DateTimeFormatter.ISO_DATE);
```

Comment:
as the referenceDate is constant at runtime its better to make it `private static final` and instantiate it once.
Also its cleaner to use a  LocalDate creator like: `LocalDate.of(2012, 8, 6);` in this case.

Reply:
Fixed this by adding 
private static final LocalDate referenceDate = LocalDate.of(2012, 8, 6);